From 941acb40b4801c714e53753a5122408af3509f2c Mon Sep 17 00:00:00 2001
From: 4m1g0 <casadelblanco@gmail.com>
Date: Sat, 7 Jan 2012 00:33:01 +0100
Subject: [PATCH 09/22] Server-Info

---
 src/server/shared/SystemConfig.h |    2 +-
 2 files changed, 4 insertions(+), 3 deletions(-)

diff --git a/src/server/shared/SystemConfig.h b/src/server/shared/SystemConfig.h
index 2431ebb..348a626 100755
--- a/src/server/shared/SystemConfig.h
+++ b/src/server/shared/SystemConfig.h
@@ -24,7 +24,7 @@
 #include "Define.h"
 #include "revision.h"
 
-#define _PACKAGENAME "TrinityCore"
+#define _PACKAGENAME "ServerWoW Core"
 
 #if TRINITY_ENDIAN == TRINITY_BIGENDIAN
 # define _ENDIAN_STRING "big-endian"
-- 
1.7.5.4

