From 5e019027b72ed3fd7c898a80a4231cce9db5c5fa Mon Sep 17 00:00:00 2001
From: 4m1g0 <casadelblanco@gmail.com>
Date: Mon, 5 Mar 2012 20:40:21 +0100
Subject: [PATCH 09/27] ServerInfo

---
 src/server/shared/SystemConfig.h |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/server/shared/SystemConfig.h b/src/server/shared/SystemConfig.h
index 2431ebb..348a626 100755
--- a/src/server/shared/SystemConfig.h
+++ b/src/server/shared/SystemConfig.h
@@ -24,7 +24,7 @@
 #include "Define.h"
 #include "revision.h"
 
-#define _PACKAGENAME "TrinityCore"
+#define _PACKAGENAME "ServerWoW Core"
 
 #if TRINITY_ENDIAN == TRINITY_BIGENDIAN
 # define _ENDIAN_STRING "big-endian"
-- 
1.7.5.4

