From cff2f2aae985e34a1d6e16492faaafe58356ff47 Mon Sep 17 00:00:00 2001
From: 4m1g0 <casadelblanco@gmail.com>
Date: Fri, 29 Jun 2012 18:28:25 +0200
Subject: [PATCH 09/29] ServerInfo

---
 src/server/shared/SystemConfig.h |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/server/shared/SystemConfig.h b/src/server/shared/SystemConfig.h
index 2431ebb..348a626 100755
--- a/src/server/shared/SystemConfig.h
+++ b/src/server/shared/SystemConfig.h
@@ -24,7 +24,7 @@
 #include "Define.h"
 #include "revision.h"
 
-#define _PACKAGENAME "TrinityCore"
+#define _PACKAGENAME "ServerWoW Core"
 
 #if TRINITY_ENDIAN == TRINITY_BIGENDIAN
 # define _ENDIAN_STRING "big-endian"
-- 
1.7.9.5

