From 785387d5074f5e1af4bd2b78f8f7ed6089db9577 Mon Sep 17 00:00:00 2001
From: 4m1g0 <casadelblanco@gmail.com>
Date: Mon, 21 May 2012 22:20:50 +0200
Subject: [PATCH 20/25] ArenaLog

---
 src/server/game/Battlegrounds/Battleground.cpp |    4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/server/game/Battlegrounds/Battleground.cpp b/src/server/game/Battlegrounds/Battleground.cpp
index 9d5858e..f83fe92 100755
--- a/src/server/game/Battlegrounds/Battleground.cpp
+++ b/src/server/game/Battlegrounds/Battleground.cpp
@@ -782,6 +782,10 @@ void Battleground::EndBattleground(uint32 winner)
                     for (Battleground::BattlegroundScoreMap::const_iterator itr = GetPlayerScoresBegin(); itr != GetPlayerScoresEnd(); ++itr)
                         if (Player* player = ObjectAccessor::FindPlayer(itr->first))
                             sLog->outArena("Statistics match Type: %u for %s (GUID: " UI64FMTD ", Team: %d, IP: %s): %u damage, %u healing, %u killing blows", m_ArenaType, player->GetName(), itr->first, player->GetArenaTeamId(m_ArenaType == 5 ? 2 : m_ArenaType == 3), player->GetSession()->GetRemoteAddress().c_str(), itr->second->DamageDone, itr->second->HealingDone, itr->second->KillingBlows);
+                
+                // Guardamos los de arena en tabla logs
+                CharacterDatabase.PExecute("INSERT INTO `logs` (type, param1, param2, param3, param4, param5, param6) VALUES (2, %u, %d, %d, %d, %d, %d)", m_ArenaType, winner_arena_team->GetId(), loser_arena_team->GetId(), winner_team_rating, loser_team_rating, GetStartTime());
+                
             }
             // Deduct 16 points from each teams arena-rating if there are no winners after 45+2 minutes
             else
-- 
1.7.9.5

