From 57788103ee756ceace0fbf4760b04defdec6c17e Mon Sep 17 00:00:00 2001
From: 4m1g0 <casadelblanco@gmail.com>
Date: Sun, 19 Feb 2012 23:22:30 +0100
Subject: [PATCH 27/27] EsferaDeHieloLichKing

---
 .../TargetedMovementGenerator.cpp                  |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/server/game/Movement/MovementGenerators/TargetedMovementGenerator.cpp b/src/server/game/Movement/MovementGenerators/TargetedMovementGenerator.cpp
index 964b440..06dbf7d 100755
--- a/src/server/game/Movement/MovementGenerators/TargetedMovementGenerator.cpp
+++ b/src/server/game/Movement/MovementGenerators/TargetedMovementGenerator.cpp
@@ -132,7 +132,7 @@ bool TargetedMovementGeneratorMedium<T,D>::Update(T &owner, const uint32 & time_
     }
 
     // prevent movement while casting spells with cast time or channel time
-    if (owner.IsNonMeleeSpellCasted(false, false,  true))
+    if (owner.IsNonMeleeSpellCasted(false, false,  true) && owner.GetEntry() != 36633 && owner.GetEntry() != 39305 && owner.GetEntry() != 39306 && owner.GetEntry() != 39307)
     {
         if (!owner.IsStopped())
             owner.StopMoving();
-- 
1.7.5.4

